va/va_cp_h264.h

```diff
@@ -0,0 +1,481 @@
+} VACencSequenceParameterBufferH264;
+
+/**
+ * \brief decoded reference picture marking for H.264 cenc decode  
+ * in baseline, main & high profiles.
+ *
+ * This structure holds information for \c dec_ref_pic_marking() as
+ * defined by the H.264 specification.
+ */
+typedef struct _VACencDecRefPictureMarkingBufferH264 {
+    union {
+        struct {
+            /** \brief Same as the H.264 bitstream syntax element. */
+            uint32_t no_output_of_prior_pics_flag             : 1;
+            /** \brief Same as the H.264 bitstream syntax element. */
+            uint32_t long_term_reference_flag                 : 1;
+            /** \brief Same as the H.264 bitstream syntax element. */
+            uint32_t adaptive_ref_pic_marking_mode_flag       : 1;
+            /** \brief number of decode reference picture marking. */
+            uint32_t dec_ref_pic_marking_count                : 8;;
+            /** \brief Reserved for future use, must be zero */
+            uint32_t reserved                                 : 21;
+            } bits;
+        uint32_t value;
+    } ref_pic_fields;
+    /** \brief Same as the H.264 bitstream syntax element. */
+    int32_t difference_of_pic_nums_minus1[17];
+    /** \brief Same as the H.264 bitstream syntax element. */
+    int32_t long_term_pic_num[17];
+    /** \brief Same as the H.264 bitstream syntax element. */
+    uint8_t memory_management_control_operation[17];
+    /** \brief Same as the H.264 bitstream syntax element. */
+    uint8_t max_long_term_frame_idx_plus1[17];
+    /** \brief Same as the H.264 bitstream syntax element. */
+    uint8_t long_term_frame_idx[17];
+    /** \brief Reserved bytes for future use, must be zero */
+    uint32_t va_reserved[VA_PADDING_MEDIUM];
+}VACencDecRefPictureMarkingBufferH264;
+
+/**
+ * \brief Slice parameter for H.264 cenc decode in baseline, main & high profiles.
+ *
+ * This structure holds information for \c
+ * slice_layer_without_partitioning_rbsp() and nal_unit()of the slice 
+ * as defined by the H.264 specification.
+ *
+ */
+typedef struct _VACencSliceParameterBufferH264 {
+
+     /** \brief Slice type.
+     *  Range: 0..2, 5..7, i.e. no switching slices.
+     */
+    uint8_t slice_type;
+     /** \brief Same as the H.264 bitstream syntax element. */
+    uint8_t nal_ref_idc;
+    /** \brief Indicate if this is coded slice of an IDR picture. 
+    * Same as the H.264 bitstream syntax element.*/
+    uint8_t idr_flag;
+    /** \brief Same as the H.264 bitstream syntax element. */
+    uint32_t frame_number;
+    /** \brief Indicate if this is a field or frame picture. */
+    uint8_t field_frame_flag;
+    /** \brief top field or frame POC. */
+    int32_t top_field_order_cnt;
+     /** \brief bottom field POC. */
+    int32_t bottom_field_order_cnt;
+    /** \brief Reserved bytes for future use, must be zero */
+    uint32_t va_reserved[VA_PADDING_MEDIUM];
+} VACencSliceParameterBufferH264;
```

Comment:
Quite a bit of stuff is missing here.  How does the user determine which reference pictures are needed without the ref pic list information?

Reply:
ref_pic_marking is returned so that DPB can be handled by user.